webapp/src/main/resources/public/js/components/workbench/GitBlameInfoModal.js

```diff
@@ -173,34 +165,56 @@ let GitBlameInfoModal = React.createClass({
             return null;
         }
 
+        // if no usages, use assetPath
         if (usages.length === 0) {
             usages = [textUnit.getAssetPath()];
         }
 
         let links = usages.join('\n');
 
-        if (this.props.appConfig.opengrok.server !== null) {
-            links = this.getOpenGrokLinks(usages);
+        if (this.props.appConfig.link.repositoryMapping !== null) {
+            let tag = this.props.textUnit.getName().split(this.props.appConfig.link.tagDelimiter)[0];
+            links = this.getLocationLinks(usages, tag);
         }
 
         return links;
     },
 
-    getOpenGrokLinks(usages) {
+    /**
+     * @param usages      Usages in code of text unit
+     * @param tag         Tag that may be used to create link
+     * @returns {*|Array} A list of links generated based on usages
+     */
+    getLocationLinks(usages, tag) {
         let repo = this.props.textUnit.getRepositoryName();
         let links = [];
+        let linkConfig = this.props.appConfig.link;
+
         for (let usage of usages) {
-            usage = usage.replace(this.props.appConfig.opengrok.extractedFilePrefix, "");
-            let link = this.props.appConfig.opengrok.server;
-            for (let repoKey in this.props.appConfig.opengrok.repositoryMapping){
-                if (repo === repoKey) {
-                    link += this.props.appConfig.opengrok.repositoryMapping[repoKey];
-                    break;
-                }
-            }
-            link += usage.replace(":", "#");
+            usage = usage.replace(this.props.appConfig.link.extractedFilePrefix, "");
```

Comment:
Add LinkHelper class to format link with basically 1 function to generate the final link

let's look at using this https://lodash.com/docs/4.17.10#template instead of `formatMessage`